File: backend/financial/financial/src/main/java/com/passamv/financial/entity/Content.java
package com.passamv.financial.entity;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity(name = "CONTENT")
public class Content {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "content_id")
    private int id;

    private String name;

    /*@Lob
    private byte[] file;*/

    private String file;

    @Column(name = "color_name")
    private String colorName;

    @Column(name = "title")
    private String title;

    @ManyToOne
    @JoinColumn(name = "module_id", referencedColumnName = "module_id")
    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
    @JsonIdentityReference(alwaysAsId = true)
    private Module module;

    public Content() {}

    public Content(int id, String name, String file, String colorName, String title, Module module) {
        this.id = id;
        this.name = name;
        this.file = file;
        this.colorName = colorName;
        this.title = title;
        this.module = module;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getFile() {
        return file;
    }

    public void setFile(String file) {
        this.file = file;
    }

    public Module getModule() {
        return module;
    }

    public void setModule(Module module) {
        this.module = module;
    }

    public String getColorName() {
        return colorName;
    }

    public void setColorName(String colorName) {
        this.colorName = colorName;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: frontend/passamv/src/app/home/dto/content.ts
import { Module } from "./module";


export class Content {
    id: number;
    name: string;
    file: string;
    title: string;
    colorName: string;
    module: Module;

    public getId() {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getName() {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }

    public getFile() {
        return this.file;
    }

    public setFile(file: string) {
        this.file = file;
    }

    public getTitle() {
        return this.title;
    }

    public setTitle(title: string) {
        this.title = title;
    }

    public getColorName() {
        return this.colorName;
    }

    public setColorName(colorName: string) {
        this.colorName = colorName;
    }

    public getModule() {
        return this.module;
    }

    public setModule(module: Module) {
        this.module = module;
    }

}